src/main/java/service/FruitTransactionParser.java

```diff
@@ -0,0 +1,21 @@
+package service;
+
+import java.util.ArrayList;
+import java.util.List;
+import model.FruitTransaction;
+import model.OperationType;
+
+public class FruitTransactionParser {
+    public List<FruitTransaction> parse(List<String> lines) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : lines) {
+            String[] data = line.split(",");
+            OperationType operation = OperationType.fromCode(data[0]);
+            String fruit = data[1];
+            int quantity = Integer.parseInt(data[2]);
+            FruitTransaction transaction = new FruitTransaction(operation, fruit, quantity);
+            transactions.add(transaction);
+        }
+        return transactions;
+    }
```

Comment:
The method parse() should include checks for possible issues with the input data, such as incorrect formatting, missing values, or invalid operation codes. If any of these issues are encountered, the method should throw a RuntimeException with an informative message to prevent invalid data from being processed.